sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -281,42 +281,54 @@ class StreamExecution(
         // Unblock `awaitInitialization`
         initializationLatch.countDown()
 
-        triggerExecutor.execute(() => {
```

Comment:
It's hard to see from the patch, but all that's happening here is that the execution is now wrapped in `withNewExecutionID`. I also had to create an `IncrementalExecution` to pass to that method, which is like a normal one, but uses `currentBatchId=-1` (the default).